document-generator-company-report/src/main/java/uk/gov/companieshouse/document/generator/company/report/mapping/mappers/recentfilinghistory/ApiToAssociatedFilings.java

```diff
@@ -0,0 +1,105 @@
+import uk.gov.companieshouse.api.model.filinghistory.AssociatedFilingsApi;
+import uk.gov.companieshouse.document.generator.common.descriptions.RetrieveApiEnumerationDescription;
+import uk.gov.companieshouse.document.generator.company.report.mapping.model.document.items.recentfilinghistory.items.AssociatedFilings;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@RequestScope
+@Mapper(componentModel = "spring")
+public abstract class ApiToAssociatedFilings {
+
+    @Autowired
+    private RetrieveApiEnumerationDescription retrieveApiEnumerationDescription;
+
+    private static final String FILING_HISTORY_DESCRIPTIONS = "FILING_HISTORY_DESCRIPTIONS";
+    private static final String STATEMENT_OF_CAPITAL = "statement-of-capital";
+    private static final String YYYY_MM_DD = "yyyy-MM-dd";
+
+    public abstract AssociatedFilings apiToAssociatedFilings(AssociatedFilingsApi associatedFilingsApi);
+
+    public abstract List<AssociatedFilings> apiToAssociatedFilings(List<AssociatedFilingsApi> associatedFilingsApi);
+
+    @AfterMapping
+    protected void convertAssociatedFilingsDescription(AssociatedFilingsApi associatedFilingsApi,
+                                                       @MappingTarget AssociatedFilings associatedFilings){
+
+        associatedFilings.setDescription(setAssociatedFilingsDescription(associatedFilingsApi.getDescription(),
+            associatedFilingsApi.getDescriptionValues()));
+    }
+
+    private String setAssociatedFilingsDescription(String description, Map<String, Object> descriptionValues) {
+
+        if (!description.equals(STATEMENT_OF_CAPITAL)) {
+            return descriptionValues.get("description").toString();
+        } else {
+
+        String associatedFilingsDescription = (retrieveApiEnumerationDescription
+            .getApiEnumerationDescription(FILING_HISTORY_DESCRIPTIONS, "description",
+                description, getDebugMap(description))) + "\r" +
+                getStatementOfCapitalDescriptionValues(descriptionValues);
+
+        if (descriptionValues != null) {
+            return populateParameters(associatedFilingsDescription, descriptionValues);
+        } else
+            return associatedFilingsDescription;
+        }
+    }
+
+    private String populateParameters(Object description, Map<String, Object> parameters) {
+
+        formatDateParameters(parameters);
+        StrSubstitutor sub = new StrSubstitutor(parameters, "{", "}");
+
+        return sub.replace(description);
+    }
+
+    private String getStatementOfCapitalDescriptionValues(Map<String, Object> descriptionValues) {
+
+        if (descriptionValues != null) {
+            List<Map<String, Object>> list = (List) descriptionValues.get("capital");
+            String valueSentence = list.get(0).get("currency").toString() + " " + list.get(0).get("figure").toString();
+
+            return valueSentence;
```

Comment:
Local variable not required:
`return list.get(0).get("currency").toString()
                + " " + list.get(0).get("figure").toString();`